test/e2e/clusterctl_upgrade.go

```diff
@@ -278,7 +278,7 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 
 			framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
 				Lister:    managementClusterProxy.GetClient(),
-				Namespace: managementClusterNamespace.Name,
+				Namespace: testNamespace.Name,
```

Comment:
Without this fix we don't get any resources from the cluster where we test the provider upgrade

Reply:
@sbueringer thanks for quick fix,

would it make sense to add test cases 
that checks the fields before and after upgrade, even for future upgrades. 
This way we can be sure we don't miss any fields across upgrades

Reply:
I think its hard to catch those kind of errors in our e2e tests:
* The test runs exactly as before even with the fix, because we're using the default value of rolloutStrategy. So you won't see any difference through the fix.
* We would have to change the value of this specific field to something which is not default (afaik in this case I would have to set maxSurge to 0 and also KCP replicas to 3) to catch the error
* Also it's hard to check for something which is missing (when you're not aware of it), i.e. in this case the v1alpha3 structs on main didn't have the field

So I guess we could modify our e2e test so it would catch this specific case of the error (although it's not trivial and not something we usually do in our e2e tests). But not sure how to do this for arbitrary fields which might be missing, when we're not aware of it.

Usually we test these kind of things in our conversion unit tests. But they also cannot catch this error if the structs on the current branch don't have the field.

So I'm not sure if there's an easy way to add a test for something like this. It essentially comes down to we have to keep the v1alpha3 structs between main and release-0.3 sync and then our conversion unit tests on main can catch errors in the conversion

Hope this makes sense :)

Reply:
Agree, keeping types in sync would have better chance to catch these errors